enhancements/windows-containers/disconnected-support.md

```diff
@@ -0,0 +1,390 @@
+  such that it enables Windows nodes to pull images in disconnected environments
+* Configure the mirror registry settings in the containerd runtime, a WMCO-managed component on Windows nodes 
+* React to changes to the mirror registry settings during WMCO runtime
+* Maintain normal functionality in connected clusters and clusters without any cluster-wide mirror registry configuration
+
+### Non-Goals
+
+* First-class support of [all OpenShift image registry setting configurations](https://docs.openshift.com/container-platform/4.14/openshift_images/image-configuration.html#images-configuration-file_image-configuration), including the `Image` resource
+* Supporting clusters that use only `ImageContentSourcePolicy` resources. These were [deprecated in 4.13](https://docs.openshift.com/container-platform/4.14/release_notes/ocp-4-14-release-notes.html#ocp-4-14-deprecated-removed-features) 
+and users [should migrate to IDMSes](https://docs.openshift.com/container-platform/4.14/openshift_images/image-configuration.html#images-configuration-registry-mirror-convert_image-configuration) instead.
+* Using [`ImageContentPolicy` resources](https://docs.openshift.com/container-platform/4.14/rest_api/config_apis/imagecontentpolicy-config-openshift-io-v1.html)
+
+
+## Proposal
+
+### Design Details
+
+WMCO will use the mirrors data in `ImageDigestMirrorSet` and `ImageTagMirrorSet` resources 
+to create a set of [hosts.toml files](https://github.com/containerd/containerd/blob/main/docs/hosts.md#hoststoml-content-description---detail),
+one for each registry mirror, that will be transferred to the Windows instance.
+
+This will be done on during initial node configuration, similarly to how bootstrap files (like the kubelet config file)
+are generated. This way nodes will have the right mirror registry config right when they join the cluster.
+
+
+In addition, the operator should also react to any events on IDMS/ITMS resources through a new "registry controller" --
+when mirror settings are updated, WMCO will then react by:
+- generating an updated set of hosts.toml files
+- publishing them to the windows-services ConfigMap as `Files` 
+
+The windows-services ConfigMap spec will have to be extended:
+* we will add a `content` field the `files` key so it will look like
+```json
+{
+  "path": "filepath",
+  "checksum": "checksum of file, generated at compile time",
+  "content": "base64 encoded file data, similarly to how MCO stores files in MachineConfigs",
+}
+```
+
+WICD, which already watches the services ConfigMap, will react to events on it by copying over the new registry config
+files to each Windows instance, if a file has changed.
+Note that if the newly generated set of files is empty, this implies the cluster mirror regsitry settings have been cleared.
+WMCO will respond by deleting all directories and hosts.toml files under containerd's `config_path` directory.
```

Comment:
This sounds like its special casing the config_path directory.
If that is the case, please avoid this by altering the windows-services configmap schema to allow a way to express what you need to do.

Reply:
Introduced "FilesToDelete" to address this